pkg/action/action.go

```diff
@@ -210,14 +222,144 @@ func (c *Configuration) renderResources(ch *chart.Chart, values chartutil.Values
+		return nil, errors.Wrap(err, "error while running post render on files")
+	}
+
+	postProcessedFiles := make(map[string]string)
+	// Split the post-processed stream into files.
+	for _, document := range reYamlDocumentSeparator.Split(b.String(), -1) {
+		if strings.TrimSpace(document) == "" {
+			continue
+		}
+		document, fileName, err := yamlCutTemporaryAnnotation(document)
 		if err != nil {
-			return hs, b, notes, errors.Wrap(err, "error while running post render on files")
+			return nil, fmt.Errorf("cannot reconstruct a file after post-process: %w", err)
+		}
+		existingDocument, ok := postProcessedFiles[fileName]
+		if ok {
+			postProcessedFiles[fileName] = existingDocument + "\n---\n" + string(document)
+		} else {
+			postProcessedFiles[fileName] = string(document)
 		}
 	}
 
-	return hs, b, notes, nil
+	return postProcessedFiles, nil
+}
+
+// yamlSetTemporaryAnnotation adds temporary annotation in yaml kubernetess object
+func yamlSetTemporaryAnnotation(document, annotationValue string) ([]byte, error) {
+	// Unmarshal the document.
+	var documentMap map[string]interface{}
+	err := yaml.Unmarshal([]byte(document), &documentMap)
+	if err != nil {
+		return nil, err
+	}
+
+	// Take the annotations map.
+	metadata, ok := documentMap["metadata"].(map[string]interface{})
+	if !ok {
+		// Document has no metadata.
+		metadata = make(map[string]interface{})
+	}
+
+	annotations, ok := metadata["annotations"].(map[string]interface{})
+
+	if !ok {
+		annotations = make(map[string]interface{})
+	}
+	annotations[tempAnnotationName] = annotationValue
+
+	metadata["annotations"] = annotations
+	documentMap["metadata"] = metadata
+
+	documentYaml, err := yaml.Marshal(documentMap)
+	if err != nil {
+		return nil, fmt.Errorf("cannot marshal a yaml document: %v", err)
+	}
+
+	return documentYaml, nil
+}
+
+// yamlCutTemporaryAnnotation cuts out an annotation text from yaml kubernetess object
+func yamlCutTemporaryAnnotation(document string) ([]byte, string, error) {
+	// Unmarshal the document.
+	var documentMap map[string]interface{}
+	err := yaml.Unmarshal([]byte(document), &documentMap)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Take the annotations map.
+	metadata, ok := documentMap["metadata"].(map[string]interface{})
+	if !ok {
+		return nil, "", fmt.Errorf("document has no metadata: %s", document)
+	}
+
+	annotations, ok := metadata["annotations"].(map[string]interface{})
+
+	annotationValue := ""
+	if ok {
+		// Cut annotation.
+		annotationValue, ok = annotations[tempAnnotationName].(string)
+		if !ok {
+			return nil, "", fmt.Errorf("annotation %s value is not a string: %v", tempAnnotationName, annotations[tempAnnotationName])
+		}
+		delete(annotations, tempAnnotationName)
+	} else {
+		// Nothing to remove. Either post-renderer removed the annotation
+		// or a new k8s object was added without such annotation.
+		// Let's add a fake file name.
+		annotationValue = fmt.Sprintf("non-existent-%d.yaml", rand.Intn(9999))
```

Comment:
This should probably use a hash of the data or a UID of some kind. It is unlikely, but you can still have a chance of a number in a random set repeating and causing a collision when trying to write out files later on

Reply:
sha1 is used instead.